Circle and Cross

atrzewik opened the pull request.

atrzewik committed:
New task

Czeffik commented on CircleAndCross.py:
```diff
@@ -0,0 +1,115 @@
+class Board(object):
+
+    def __init__(self):
```
`Board` ma sie skladac z dwuwymiarowej listy `fields`, ktora sklada sie z pol `Field`. Ewentulanie w konstruktorze mozesz dodac jeszcze funkcje `setup_fields` jesli bedziesz tego potrzebowala

Czeffik commented on CircleAndCross.py:
```diff
@@ -0,0 +1,115 @@
+            board.append(["-"]*3)
+        return board
+
+    def print_board(self, board):
```
ta funkcja bedzie potrzebna! bedzie printowac dwuwymiarowa liste `fields`. Ewentualnie te funkcjonalnosc mozna przeniesc do funkcji `__repr__` lub `__str__`

Czeffik commented on CircleAndCross.py:
```diff
@@ -0,0 +1,115 @@
+                    list_of_signs.append([i, j])
+        return list_of_signs
+
+    def check_the_winner(self, sign, board):
```
taka funkcja bedzie potrzebna. Ale musisz zrobic algorytmy sprawdzajace zwycieztwo. Polecam w funkcji `check_winner` wywolywac nastepujace funkcje: 

`row_winning_conditio`
`column_winning_condition`
`diagonal_top_condition`
`diagonal_down_condition`

wszystkie te funkcje beda zwracaly `boolean` to znaczy (True/False)

a funkcja `check_winner` bedzie kombinacja tych wszystkich metod: to jest jesli jedna z tych metod zwroci true, to znaczy, ze  ktos wygral

Czeffik commented on CircleAndCross.py:
```diff
@@ -0,0 +1,115 @@
+        self.circle = Field().circle
+        self.cross = Field().cross
+
+    def row_and_column(self):
```
niepotrzebne

Czeffik commented on CircleAndCross.py:
```diff
@@ -0,0 +1,115 @@
+    def __init__(self):
+        self.board = self.row_and_column()
+        self.printed_board = self.print_board(self.board)
+        self.circle = Field().circle
```
pozostale rzeczy sa niepotrzebne w konstruktorze

Czeffik commented on CircleAndCross.py:
```diff
@@ -0,0 +1,115 @@
+            new_board += " ".join(row) + "\n"
+        return new_board
+
+    def add_to_board(self, row, column, sign, board):
```
taka funkcja jest niepotrzebna. Ale bedziemy potrzebowac czegos w stylu `make_move`, w ktorej to funkcji bedziemy przyjmowac `row_number` i `column_number` oraz rodzaj znacznika (kolko/krzyzyk). Pamietaj tutaj o walidacji, ze pole musi byc puste, zeby moc zaznaczyc, jesli nie jest puste, to rzuc jakis `Exception` (IllegalArgumentException albo cos w tym stylu)

Czeffik commented on CircleAndCross.py:
```diff
@@ -0,0 +1,115 @@
+        board[int(row)][int(column)] = sign
+        return self.print_board(board)
+
+    def co_ordinates_of_signs(self, sign, board):
```
wywal

Czeffik commented on CircleAndCross.py:
```diff
@@ -0,0 +1,115 @@
+
+class Field(object):
+
+    def __init__(self):
```
tutaj potrzebujesz przechowywac tylko informacje o tym, co jest na polu. Czy jest to `X`/`O`/ `-`.
Zastanow sie jak jest Ci to najlatwiej przechowywac. Mozlwe, ze pole bedzie nazywac sie `state`.

Czeffik commented on CircleAndCross.py:
```diff
@@ -0,0 +1,115 @@
+        self.circle = self.circle_sign()
+        self.cross = self.cross_sign()
+
+    def circle_sign(self):
```
wywal, ale wytlumacz mi co to mialo robic XD co oznacza `'o' or 'O'`?

Czeffik commented on CircleAndCross.py:
```diff
@@ -0,0 +1,115 @@
+        circle = "o" or "O"
+        return circle
+
+    def cross_sign(self):
```
wywal

Czeffik commented on CircleAndCross.py:
```diff
@@ -0,0 +1,115 @@
+
+class Game(object):
+
+    def __init__(self):
```
`Game` bedzie sie skladal z pola `board` ktore bedzie po prostu `Board()` em 😄 bardzo prawdopodobne, ze przyda Ci sie tez informacja o tym ktora jest tura: `turn_number` oraz jaka jest pojemnosc stolu `board_capacity` - to bedzie potrzebne, zeby gra sie zakonczyla gdy juz nie ma wolnego miejsca na stole.

Reszta rzeczy do wywalenia

Czeffik commented on CircleAndCross.py:
```diff
@@ -0,0 +1,115 @@
+        self.circle = Field().circle_sign()
+        self.cross = Field().cross_sign()
+
+    def moves(self):
```
ta funkcja bedzie do rozbicia na mniejsze pomocnicze

Czeffik commented on CircleAndCross.py:
```diff
@@ -0,0 +1,115 @@
+                    else:
+                        continue
+        return winner
+
```
Bedziemy potrzbowac funkcji, ktora bedzie sprawdzala czy kontynuujemy gre,np. `continue_game`, ktora bedzie sprawdzac czy `self.board.check_wineer` jest `True`, albo czy `turn_number == board_capacity`, bo wtedy bedzie remis, jesli nikt nie zdazyl wygrac.

Czeffik commented on CircleAndCross.py:
```diff
@@ -0,0 +1,115 @@
+                        continue
+        return winner
+
+
```
dodatkowo potrzebujemy fukcji `make_move`, ktora przy udanym ruchu bedzie zwiekszala `turn-number`, a przy ruchu nieudanym - np. w zajete pole, bedzie zwracala informacje, ze gracz cos zepsul i zeby zrobil ruch ponownie, oraz wyprintuje stol(stol printujesz takze po udanym ruchu, zeby gracze wiedzieli jaki jest aktualny uklad)

Czeffik replied:
bedziesz tez potrzebowala ustalic czyj jest ruch, czy kolko czy krzyzk. Zrob do tego funkcje i wykorzystaj `turn_number`

Czeffik replied:
oraz sam mechanizm gry,  ktory bedziesz powtarzala tak dlugo, az jakis gracz wygra, albo skonczy sie miejsce na planszy.

w tej funkcji bedziesz pobierala sobie informacje czyj jest ruch. printowala uzytkownikowi stol oraz informacje, ze teraz jest tura gracza `X`/`O`. Prosila o podanie rowa, o podanie columny, a nastepnie wykonywala ruch. Zaznaczala go na planszy ;)

Czeffik reviewed (changes requested):
Mozliwe, ze duzo latwiej bedzie Ci zrezygnowac z klasy `Field` i wtedy stol bedzie sie skladac z dwuwymiarowej listy Stringow. Chyba powinnas zaczac od tego podejscia. Dodawanie Field'a mocno komplikuje sprawe 😄 

GOOD LUCK!

atrzewik committed:
Total changes

atrzewik committed:
make it finally objective

Czeffik reviewed (approved):
Testy przesnies do innego pliku. Dodaj klase (najlepiej w innym pliku) do przechwytywania inputu od uzytkownika. Zastanow sie nad logika sprawdzania wygranej w klasie Board. Wszystkie funkcje powinny operowac na wartosciach. Input uzytkownika bedzie w innym miejscu 😄

atrzewik committed:
Changed logic in checking winner function

atrzewik committed:
reset

atrzewik committed:
reset

atrzewik committed:
Another logic

Czeffik commented on CircleAndCross.py:
```diff
@@ -22,47 +22,41 @@ def mark_field(self, row, column, sign):
-    def co_ordinates_of_signs(self, sign):
-        list_of_signs = []
+    def row_winning_condition(self, sign):
         for i in range(self.number_of_fields):
```
co to jest i, co to jest j?

Czeffik replied:
może by tak nazwać to, row_number i column_number, albo coś w ten deseń?

Czeffik commented on CircleAndCross.py:
```diff
@@ -22,47 +22,41 @@ def mark_field(self, row, column, sign):
-        return self.check_winning_condition(winning, sign)
+            count = 0
+            for j in range(self.number_of_fields):
+                if self.fields[j][(self.number_of_fields - 1)-j] == sign:
```
`(self.number_of_fields - 1)-j` może by to wyciągnąć do zmiennej i jakoś nazwać? ja na to patrząc nie wiem co sie tutaj dzieje

atrzewik committed:
Cleaning

atrzewik committed:
Add file with user input, but when given input is wrong, NoneType data ruing everything

atrzewik committed:
User Input added static method.
My User Input correct by add "return", now its also working.
Added all winning condition to testing in another file.
Circle And Cross update with static User Input class.
